src/snowflake/snowpark/types.py

```diff
@@ -982,6 +1047,7 @@ def _fill_ast(self, ast: proto.DataType) -> None:
 
 _FIXED_VECTOR_PATTERN = re.compile(r"vector\(\s*(int|float)\s*,\s*(\d+)\s*\)")
 _FIXED_DECIMAL_PATTERN = re.compile(r"decimal\(\s*(\d+)\s*,\s*(\d+)\s*\)")
+_INTERVAL_YEARMONTH_PATTERN = re.compile(r"interval (year|month)( to (year|month))?")
```

Comment:
```suggestion
_INTERVAL_YEARMONTH_PATTERN = re.compile(r"interval (year|month)( to month)?")
```

We are always guaranteed that is it one of the following formats (haven't included precision or the actual value itself):

INTERVAL YEAR
INTERVAL MONTH
INTERVAL YEAR TO MONTH

Reply:
So INTERVAL MONTH TO YEAR is not supported? I noticed that I could do this in PySpark but admittedly didn't do much further testing

```>>> a = YearMonthIntervalType(1, 0)
>>> a
YearMonthIntervalType(1, 0)
>>> a.__repr__()
'YearMonthIntervalType(1, 0)'
>>> >>> a.simpleString()
'interval month to year'
```

Reply:
Yes if the leading field is a lesser unit that the trailing field (e.g. INTERVAL MONTH TO YEAR), that violates ANSI SQL standard so isn't supported in most engines (Oracle, Teradata, Snowflake, Trino etc.). Not sure of the PySpark behavior but it is invalid in Snowflake.